File: C/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int countLetters(string text);
int countWords(string text);
int countSentences(string text);
int main(void){
    string text = get_string ("Text: ");
    int letters = countLetters(text);
    int words = countWords(text);
    int sentences = countSentences(text);
    float L = (letters *  100.0) / words;
    float S = (sentences * 100.0) / words;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    if (index < 1){
        printf("Before Grade 1\n");
    }
    else if (index >= 16){
        printf("Grade 16+\n");
    }
    else{
        printf("Grade %i\n", (int)round(index));
    }
}



//to count number of letters
int countLetters(string text){
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i ++){
        if (isalpha(text[i])){
            letters += 1;
        }
    }
    return letters;
}


//to count number of words
int countWords(string text){
    int words = 0;
    for (int i = 0, n = strlen(text); i < n; i ++){
        if (isspace(text[i])){
            words += 1;
        }
    }
    return words + 1;
}


//to count number of sentences
int countSentences(string text){
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i ++){
        if (text[i] == '.' || text[i] == '?' || text[i] == '!'){
            sentences += 1;
        }
    }
    return sentences;
}

File: Python/readability.py
# Here's a little Easter egg showing that I, Ezekwe Gideon Somtochukwu actually wrote this code. IG:@lets_patois
from cs50 import get_string

def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Implement Coleman-Liau formula
    L = (letters *  100.0) / words
    S = (sentences * 100.0) / words
    index = 0.0588 * L - 0.296 * S - 15.8

    # Print out the grades according to the index
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")



def count_letters(text):
    # Initialise the counter
    count = 0

    # Loop through and count each alphabet character.
    for char in text:
        if char.isalpha():
            count += 1
    return count

def count_words(text):
    # Split technique learnt from Programming with Mosh (YouTube channel)
    return len(text.split())


def count_sentences(text):
    count = 0

    # Loop through and count terminal punctuations
    for char in text:
        if char in [".", "?", "!"]:
            count += 1
    return count

# Calling the main function
main()
